src/main/java/fact/io/hdureader/FitsStream.java

```diff
@@ -0,0 +1,111 @@
+package fact.io.hdureader;
+
+import stream.Data;
+import stream.data.DataFactory;
+import stream.io.AbstractStream;
+import stream.io.SourceURL;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The ZFitsStream can read FITS files containing raw data as recorded by the FACT telescope DAQ.
+ * It also reads FITS files written by the Monte Carlo simulation program CERES.
+ * It can decompress binary tables which have been compressed according to
+ * the ZFITS standard. See http://arxiv.org/abs/1506.06045.
+ *
+ * Created by mackaiver on 12/12/16.
+ */
+public class FitsStream extends AbstractStream {
+
+    private short[] offsetCalibrationsConstants;
+    private Reader reader;
+
+    private HDU eventHDU;
+    private Map<String, Serializable> fitsHeader = new HashMap<>();
+
+    public FitsStream(SourceURL url){
+        this.url = url;
+    }
+
+    @Override
+    public void init() throws Exception {
+        super.init();
+
+        //create a fits object from the SourceURL this AbstractStream contains.
+        //The FITs object does not know SourceURL so this fits reader is decoupled entirely from the streams framework
+        URL url = new URL(this.url.getProtocol(), this.url.getHost(), this.url.getPort(), this.url.getFile());
+        Fits fits = new Fits(url);
+
+        //get calibration constants they are stored in the first (and only) row of this hdu.
+        HDU offsetHDU = fits.getHDU("ZDrsCellOffsets");
+        if (offsetHDU != null){
+            OptionalTypesMap<String, Serializable> row = ZFitsHeapReader.forTable(offsetHDU.getBinTable()).getNextRow();
+
+            offsetCalibrationsConstants = row
+                    .getShortArray("OffsetCalibration")
+                    .orElseThrow(() -> new IOException("OffsetCalibration not found in file."));
+
+        }
+
+
+        //create a refrence to the events hdu
+        eventHDU = fits.getHDU("Events");
+        BinTable eventsTable = eventHDU.getBinTable();
+
+        //read each headerline and try to get the right datatype
+        //from smallest to largest datatype
+        //if no number can be found simply save the string.
+        Header header = eventHDU.header;
+        fitsHeader = header.asMapOfSerializables();
+
+        Boolean ztable = eventHDU.header.getBoolean("ZTABLE").orElse(false);
+        if (ztable) {
+            reader = ZFitsHeapReader.forTable(eventsTable);
+        } else {
+            reader = BinTableReader.forBinTable(eventsTable);
+        }
+    }
+
+
+
+    private void applyDrsOffsetCalib(int numSlices, short[] data, short[] startCellData, short[] calibrationConstants){
```

Comment:
One can save time here by not adding these constants here at all but instead adding them together with the other constants in the "other" DRS amplitude calibration which comes after reading the raw data. 
Not sure how much time it saves though. 
However, by not adding these constants, this "raw data" is already "pseudo calibrated" which means, people can already see single pulses on this pseudo calibrated data. Entirely for free.
By adding these constants, one just adds noise to the timeline, i.e. one scrambles the raw data ... which costs time and doesn't add value. 

One does "restore" the raw data, yes ... but who is interested in that noise raw data? 
I never met anybody.